ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/integration/SystemAuthenticationFunctionalTest.java

```diff
@@ -0,0 +1,331 @@
+      .put("authentication.roles", "**")
+      // Reuse the Kafka JAAS config for KSQL authentication which has the same valid users
+      .put("authentication.realm", JAAS_KAFKA_PROPS_NAME)
+      .put(
+          KsqlConfig.KSQL_SECURITY_EXTENSION_CLASS,
+          MockKsqlSecurityExtension.class.getName()
+      )
+      .build();
+
+  private static final Map<String, Object> COMMON_CONFIG = ImmutableMap.<String, Object>builder()
+      .put(KsqlRestConfig.KSQL_HEARTBEAT_ENABLE_CONFIG, true)
+      .put(KsqlRestConfig.KSQL_HEARTBEAT_SEND_INTERVAL_MS_CONFIG, 200)
+      .put(KsqlRestConfig.KSQL_HEARTBEAT_CHECK_INTERVAL_MS_CONFIG, 1000)
+      .put(KsqlRestConfig.KSQL_HEARTBEAT_DISCOVER_CLUSTER_MS_CONFIG, 1000)
+      .put(KSQL_STREAMS_PREFIX + StreamsConfig.STATE_DIR_CONFIG, getNewStateDir())
+      .put(KSQL_STREAMS_PREFIX + StreamsConfig.NUM_STANDBY_REPLICAS_CONFIG, 1)
+      .put(KsqlConfig.KSQL_SHUTDOWN_TIMEOUT_MS_CONFIG, 1000)
+      .putAll(ServerKeyStore.keyStoreProps())
+      .build();
+
+  private static Map<String, String> internalKeyStoreProps(boolean node1) {
+    Map<String, String> keyStoreProps = node1
+        ? MultiNodeKeyStore.keyStoreNode1Props()
+        : MultiNodeKeyStore.keyStoreNode2Props();
+    Map<String, String> trustStoreProps = MultiNodeTrustStore.trustStoreNode1Node2Props();
+    return ImmutableMap.of(
+        SslConfigs.SSL_KEYSTORE_LOCATION_CONFIG,
+        keyStoreProps.get(SslConfigs.SSL_KEYSTORE_LOCATION_CONFIG),
+        SslConfigs.SSL_KEYSTORE_PASSWORD_CONFIG,
+        keyStoreProps.get(SslConfigs.SSL_KEYSTORE_PASSWORD_CONFIG),
+        SslConfigs.SSL_TRUSTSTORE_LOCATION_CONFIG,
+        trustStoreProps.get(SslConfigs.SSL_TRUSTSTORE_LOCATION_CONFIG),
+        SslConfigs.SSL_TRUSTSTORE_PASSWORD_CONFIG,
+        trustStoreProps.get(SslConfigs.SSL_TRUSTSTORE_PASSWORD_CONFIG)
+    );
+  }
+
+  private static final BasicCredentials USER1 = BasicCredentials.of(
+      VALID_USER1.username,
+      VALID_USER1.password
+  );
+
+  private static void commonClassSetup(final IntegrationTestHarness TEST_HARNESS,
+      final TestKsqlRestApp REST_APP_0) {
+    TEST_HARNESS.ensureTopics(2, PAGE_VIEW_TOPIC);
+    TEST_HARNESS.produceRows(PAGE_VIEW_TOPIC, PAGE_VIEWS_PROVIDER, FormatFactory.JSON);
+    RestIntegrationTestUtil.createStream(REST_APP_0, PAGE_VIEWS_PROVIDER, Optional.of(USER1));
+    RestIntegrationTestUtil.makeKsqlRequest(
+        REST_APP_0,
+        "CREATE STREAM S AS SELECT * FROM " + PAGE_VIEW_STREAM + ";",
+        Optional.of(USER1)
+    );
+  }
+
+  @RunWith(MockitoJUnitRunner.class)
+  public static class MutualAuth {
+    private static final IntegrationTestHarness TEST_HARNESS = IntegrationTestHarness.build();
+    private static final TestKsqlRestApp REST_APP_0 = TestKsqlRestApp
+        .builder(TEST_HARNESS::kafkaBootstrapServers)
+        .withEnabledKsqlClient(LOCALHOST_FACTORY)
+        .withProperty(KsqlRestConfig.LISTENERS_CONFIG, "http://0.0.0.0:8088")
```

Comment:
Avoid hardcoded ports in tests - it causes tests to fail if the ports in use, e.g. 
* If the build server is running two builds on the same box
* An engineer is running two builds, for two different change sets, at the same time.
* An engineer is running ksql to test something else, while they've also got a build running.
* etc.

I *think* you should be able to just set it to `http://0.0.0.0:0` and the server will auto-pick a free port.  It certainly did in Jetty days, not sure about Vert.x.  Check with Tim if it doesn't work. I'm sure he'll know how to.

Please update all places the port is hardcoded.